Automattic/jetpack
# CRM: WooSync order issues and invoices with the fields fee, shipping_tax, subtotals and the calculated amounts

Several fixes.

## Proposed changes:

This PR fixes multiple issues related to invoices:

- The [fee value from WooCommerce orders is not added to the invoice](https://github.com/Automattic/zero-bs-crm/issues/3017), provoking and difference between the total amount in the order and the payment object and the invoice.
- [The shipping tax from WooCommerce is not added to the invoice](https://github.com/Automattic/zero-bs-crm/issues/3023), provoking a difference between total amounts.
- [The subtotal amount from the WooCommerce order is not set to the invoice](https://github.com/Automattic/zero-bs-crm/issues/3017)
- If you update the invoice, [the shipping tax selected is removed, and the invoice is recalculated with incorrect data](https://github.com/Automattic/zero-bs-crm/issues/2923).

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:
- Install WooCommerce and load at least one product. Setting: add one tax `admin.php?page=wc-settings&tab=tax&section=standard`. Check Shipping in the tax you created.
- Enable Invoicing Settings on JPCRM: tax, discounts, shipping `admin.php?page=zerobscrm-plugin-settings&tab=invbuilder`
- WooSync Setting, active: create invoices from WooCommerce Orders `admin.php?page=zerobscrm-plugin-settings&tab=woosync`
- Create a WooCommerce order, add a product, add a fee, add a tax, add a shipping cost
![image](https://user-images.githubusercontent.com/9832440/227179384-5617b377-bf5c-409f-a844-da98441ebb4a.png)

- Recalculate the order if it's needed
- Save it
- Check the JPCRM invoice generated

**In trunk:** the amount will be different because of the fee, and the shipping tax values are absent. The calculated taxes is wrong because of the discount. In the  Preview or PDF invoice file, the subtotal value is `0.00`:

![image](https://user-images.githubusercontent.com/9832440/227515229-4869ccce-bc17-47c4-9c2a-d262dbd4a39d.png)

![image](https://user-images.githubusercontent.com/9832440/227514731-764ab867-3122-40a1-bcdc-47e9ec418bad.png)

**In this branch:** All the amounts should match:

![image](https://user-images.githubusercontent.com/9832440/227180235-96250d7c-dafc-4177-99a5-97f844f64d9f.png)

- Play by adding different items to the orders, like coupons. After updating the order, the invoice amounts should match the order amounts.
